os/overriding-systemd-units.md

```diff
@@ -0,0 +1,105 @@
+
+The advantage of the first method is that one easily overrides the complete unit, the default CoreOS unit is not parsed at all anymore. It has the disadvantage that improvements to the unit file supplied by CoreOS are not automatically incorporated on updates.
+
+The advantage of the second method is that one only overrides the settings one specifically wants, where updates to the original CoreOS unit automatically apply. This has the disadvantage that some future CoreOS updates might be incompatible with the local changes.
+
+Note that for drop-in files, if one wants to remove entries from a setting that is parsed as a list (and is not a dependency), such as ConditionPathExists= (or e.g. ExecStart= in service units), one needs to first clear the list before re-adding all entries except the one that is to be removed. See below for an example.
+
+This also applies for user instances of systemd, but with different locations for the unit files. See the section on unit load paths for further details.
+
+Let's review `fleet.service` unit /usr/lib64/systemd/system/fleet.service with the following contents:
+
+```
+[Unit]
+Description=fleet daemon
+
+After=etcd.service
+After=etcd2.service
+
+Wants=fleet.socket
+After=fleet.socket
+
+[Service]
+ExecStart=/usr/bin/fleetd
+Restart=always
+RestartSec=10s
+```
+
+Now one wants to increase `RestartSec` setting.
+
+## Override whole unit file
+
+The first possibility is to copy the unit file to `/etc/systemd/system/fleet.service` and change the chosen settings:
+
+```
+[Unit]
+Description=fleet daemon
+
+After=etcd.service
+After=etcd2.service
+
+Wants=fleet.socket
+After=fleet.socket
+
+[Service]
+ExecStart=/usr/bin/fleetd
+Restart=always
+RestartSec=60s
+```
+
+Here is how that could be implemented within `cloud-config`:
+
+```yaml
+#cloud-config
+
+coreos:
+  units:
+    - name: fleet.service
+      command: start
+      content: |
+        [Unit]
+        Description=fleet daemon
+        
+        After=etcd.service
+        After=etcd2.service
+        
+        Wants=fleet.socket
+        After=fleet.socket
+        
+        [Service]
+        ExecStart=/usr/bin/fleetd
+        Restart=always
+        RestartSec=60s
+```
+
+## Override only specific option
+
+Alternatively, the administrator could create a drop-in file `/etc/systemd/system/fleet.service.d/10-restart_60s.conf` with the following contents:
+
+```
+[Service]
+RestartSec=60s
+```
+
+And `cloud-config` example:
+
+```yaml
+#cloud-config
+coreos:
+  units:
+    - name: fleet.service
+      drop-ins:
+        - name: 10-restart_60s.conf
+          content: |
+            [Service]
+            RestartSec=60s
+      command: start
+```
+
```

Comment:
I think we need a summary here. Something about how small and targeted the change is. Maybe circle back to some of the benefits we talked about above.